winit-core/src/keyboard.rs

```diff
@@ -1707,34 +1710,39 @@ bitflags! {
 }
 
 impl ModifiersState {
-    /// Returns `true` if the shift key is pressed.
+    /// Returns whether the shift modifier is active.
     pub fn shift_key(&self) -> bool {
         self.intersects(Self::SHIFT)
     }
 
-    /// Returns `true` if the control key is pressed.
+    /// Returns whether the control modifier is active.
     pub fn control_key(&self) -> bool {
         self.intersects(Self::CONTROL)
     }
 
-    /// Returns `true` if the alt key is pressed.
+    /// Returns whether the alt modifier is active.
     pub fn alt_key(&self) -> bool {
         self.intersects(Self::ALT)
     }
 
-    /// Returns `true` if the meta key is pressed.
+    /// Returns whether the meta modifier is active.
     pub fn meta_key(&self) -> bool {
         self.intersects(Self::META)
     }
 }
 
-/// The state of the particular modifiers key.
+/// The logical state of the particular modifiers key.
 #[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum ModifiersKeyState {
-    /// The particular key is pressed.
+    /// The particular modifier is active or logically, but not necessarily physically, pressed.
     Pressed,
     /// The state of the key is unknown.
+    ///
+    /// Can also be pressed, for example, when a sticky **Shift**
+    /// is active, the OS might not include information that it was activated by RightShift, so the
+    /// [ModifiersKeyState] of [ModifiersKeys::RSHIFT] will be unknown while the [ModifiersState]
+    /// of [ModifiersState::SHIFT] will be active.
```

Comment:
Nit: s/ModifiersState/ModifiersKeyState/ in second sentence.

```suggestion
    /// [`ModifiersKeyState`] of [`ModifiersKeys::RSHIFT`] will be unknown while the
    /// [`ModifiersKeyState`] of [`ModifiersState::SHIFT`] will be active.
```

Reply:
2nd sentence refers to the side-agnostic API, so it's `ModifiersState`, different from the 1st sentence of side-aware APIs which is `ModifiersKeyState`. I've replace both first mentions with a generic `state` to avoid this confusing duplication